### ISSUE TITLE
Update nalgebra to 0.12

### ISSUE THREAD

**Issue by phrohdoh** [ChariotEngine/Chariot#11]
This will require fixing `crates/types/src/vector3.rs`:

```rust
    #[test]
    fn test_typical_normalize_use_case() {
        let position: Vector3 = Vector3::new(100.into(), 100.into(), 1.into());
        let velocity: Vector3 = Vector3::new(1.into(), 2.into(), 0.5.into());
        let time: Fixed = 0.0166666667.into();

        let new_position = position + velocity * time;
        let delta = new_position - position;
        let direction = delta.normalized();
        let direction_length = direction.length();

        assert_eq!("(100.0166667, 100.0333333, 1.0083333)".to_string(),
                   format!("{}", new_position));
        assert_eq!("(0.0166667, 0.0333333, 0.0083333)".to_string(),
                   format!("{}", delta));
        assert_eq!("(0.4364425, 0.8728850, 0.2182212)".to_string(),
                   format!("{}", direction));
        assert_eq!(1, direction_length.to_i32().unwrap());
    }
```

```
  --> src/vector3.rs:78:34
   |
78 |                    format!("{}", new_position));
   |                                  ^^^^^^^^^^^^ the trait `alga::general::real::Real` is not implemented for `fixed::Fixed`
   |
   = note: required because of the requirements on the impl of `std::fmt::Display` for `nalgebra::Matrix<fixed::Fixed, nalgebra::U3, nalgebra::U1, nalgebra::MatrixArray<fixed::Fixed, nalgebra::U3, nalgebr
a::U1>>`
   = note: required by `std::fmt::Display::fmt`

error[E0277]: the trait bound `fixed::Fixed: alga::general::real::Real` is not satisfied
  --> src/vector3.rs:80:34
   |
80 |                    format!("{}", delta));
   |                                  ^^^^^ the trait `alga::general::real::Real` is not implemented for `fixed::Fixed`
   |
   = note: required because of the requirements on the impl of `std::fmt::Display` for `nalgebra::Matrix<fixed::Fixed, nalgebra::U3, nalgebra::U1, nalgebra::MatrixArray<fixed::Fixed, nalgebra::U3, nalgebr
a::U1>>`
   = note: required by `std::fmt::Display::fmt`

error[E0277]: the trait bound `fixed::Fixed: alga::general::real::Real` is not satisfied
  --> src/vector3.rs:82:34
   |
82 |                    format!("{}", direction));
   |                                  ^^^^^^^^^ the trait `alga::general::real::Real` is not implemented for `fixed::Fixed`
   |
   = note: required because of the requirements on the impl of `std::fmt::Display` for `nalgebra::Matrix<fixed::Fixed, nalgebra::U3, nalgebra::U1, nalgebra::MatrixArray<fixed::Fixed, nalgebra::U3, nalgebr
a::U1>>`
   = note: required by `std::fmt::Display::fmt`
```

**Comment by angered-ghandi** [comment]
Looks like nalgebra admits this shortcoming in their library:
https://github.com/sebcrozet/nalgebra/blob/e0cc7ff21b0db6aa22f35bd3dbf7c696313b5f5f/src/core/matrix.rs#L858-L861

This should just work, but their Display implementation requires [Real](https://docs.rs/alga/0.5.2/alga/general/trait.Real.html), which isn't really feasible for the Fixed type that's being used.

Since this test is the only thing using Display, we could just introduce a helper method for the time being.

**Comment by phrohdoh** [comment]
Closed by #12.
